### PR TITLE
Implement legacy tab

### PR DESCRIPTION
**Pull request by Pablete1234**
The code can use a big cleanup, but it's working:
image
(pic by godout)
image
(pic by indicado)

After those screenshots, i've also fixed it so that your own name will be properly in bold

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+  // the full height minus the header, blank row, and team header.
+  private static final int MAX_TEAM_SIZE = TabDisplay.HEIGHT - 3;
+
+  // Number of players below the max before changing back to non-compact mode
+  private static final int COMPACT_MODE_HYSTERESIS = 2;
+
+  private final PGM pgm;
+  private final TabDisplay tabDisplay;
+  private BukkitTask timeUpdateTask;
+  private BukkitTask deferredRenderTask;
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java]: Can we use PGM scheduler primitives?

File: core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+  public LegacyMatchTabDisplay(PGM pgm) {
+    this.pgm = pgm;
+    this.tabDisplay = new TabDisplay(pgm, WIDTH);
+  }
+
+  public void enable() {
+    this.tabDisplay.enable();
+
+    for (Player viewer : this.pgm.getServer().getOnlinePlayers()) {
+      if (ViaUtils.getProtocolVersion(viewer) != ViaUtils.VERSION_1_7) return;
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java]: `>`

File: core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    this.timeUpdateTask.cancel();
+    this.timeUpdateTask = null;
+
+    HandlerList.unregisterAll(this);
+    this.tabDisplay.disable();
+  }
+
+  @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
+  public void onPlayerJoin(PlayerJoinEvent event) {
+    if (ViaUtils.getProtocolVersion(event.getPlayer()) != ViaUtils.VERSION_1_7) return;
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java]: `>` Same for all these checks, if someone we can allow 1.6 players then why not

File: core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+                      mapInfo.getAuthors().iterator().next().getNameLegacy(), TextColor.GRAY)),
+              viewer.getBukkit()));
+    } else {
+      this.tabDisplay.set(
+          bukkit,
+          1,
+          0,
+          TextTranslations.translateLegacy(
+              TranslatableComponent.of(
+                  "tablist.authors.tooMany",
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java]: TBH I think we can leave this out. If too many, just show the first one.

File: core/src/main/java/tc/oc/pgm/tablist/MatchTabManager.java [hunk 5]
```diff
@@ -117,7 +122,8 @@ protected void invalidate(MatchPlayer player) {
 
   @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
   public void onJoin(PlayerJoinEvent event) {
-    TabView view = this.getView(event.getPlayer());
+    if (ViaUtils.getProtocolVersion(event.getPlayer()) == ViaUtils.VERSION_1_7) return;
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/tablist/MatchTabManager.java]: `<=`

**Review by Electroid — CHANGES_REQUESTED** [review]
Some small changes. Eventually `adventure` will support legacy + modern tab lists, so I'm not too worried about refactoring later. Really great work!

File: core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java (hunk 4 above)
Comment (Pablete1234) [reply core/src/main/java/tc/oc/pgm/tablist/LegacyMatchTabDisplay.java]: Showing first author, would be unfair for the other authors on there

File: util/src/main/i18n/templates/map.properties [hunk 6]
```diff
@@ -56,6 +56,8 @@ map.setNext.revert = {0} has removed {1} as the next map
 
 map.noNextMap = No next map
 
+tablist.authors.tooMany = type {0} for info
```
Comment (Electroid) [inline util/src/main/i18n/templates/map.properties]: `map.info.commandHint = Type {0} for info`

**Review by Electroid — CHANGES_REQUESTED** [review]
Ok with the authors string, but see my comment

**Review by Electroid — APPROVED** [review]
🚀

**Pull request merged**